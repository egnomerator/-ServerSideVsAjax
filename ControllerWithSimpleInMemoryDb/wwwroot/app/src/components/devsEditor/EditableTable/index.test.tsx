import * as React from "react";
import * as ReactTestRenderer from "react-test-renderer";
import { EditableTable } from ".";

describe("EditableTable", () => {
    it("should render empty", () => {
        const devs = [];

        const table = ReactTestRenderer.create(
            <EditableTable
                devs={devs}
                view={() => { }}
                edit={() => { }}
                delete={() => { }}
            />);

        expect(table).toMatchSnapshot();
    });

    it("should render with 1 dev", () => {
        const devs = [
            { id: 1, firstName: "first1", lastName: "last1" }
        ];

        const table = ReactTestRenderer.create(
            <EditableTable
                devs={devs}
                view={() => { }}
                edit={() => { }}
                delete={() => { }}
            />);

        expect(table).toMatchSnapshot();
    });

    it("should render with 2 devs", () => {
        const devs = [
            { id: 1, firstName: "first1", lastName: "last1" },
            { id: 2, firstName: "first2", lastName: "last2" }
        ];

        const table = ReactTestRenderer.create(
            <EditableTable
                devs={devs}
                view={() => { }}
                edit={() => { }}
                delete={() => { }}
            />);

        expect(table).toMatchSnapshot();
    });

    it("should render with 3 devs", () => {
        const devs = [
            { id: 1, firstName: "first1", lastName: "last1" },
            { id: 2, firstName: "first2", lastName: "last2" },
            { id: 3, firstName: "first3", lastName: "last3" }
        ];

        const table = ReactTestRenderer.create(
            <EditableTable
                devs={devs}
                view={() => { }}
                edit={() => { }}
                delete={() => { }}
            />);

        expect(table).toMatchSnapshot();
    });
});