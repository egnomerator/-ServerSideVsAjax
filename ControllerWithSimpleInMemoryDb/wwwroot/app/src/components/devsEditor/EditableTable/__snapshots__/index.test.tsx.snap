// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableTable should render empty 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <td>
        Id
      </td>
      <td>
        Name
      </td>
      <td>
        Action
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className="text-center"
        colSpan={3}
      >
        There are no devs.
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`EditableTable should render with 1 dev 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <td>
        Id
      </td>
      <td>
        Name
      </td>
      <td>
        Action
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        1
      </td>
      <td>
        <input
          onChange={[Function]}
          type="text"
          value="first1"
        />
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          View
        </button>
         
        <button
          onClick={[Function]}
        >
          Edit
        </button>
         
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`EditableTable should render with 2 devs 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <td>
        Id
      </td>
      <td>
        Name
      </td>
      <td>
        Action
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        1
      </td>
      <td>
        <input
          onChange={[Function]}
          type="text"
          value="first1"
        />
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          View
        </button>
         
        <button
          onClick={[Function]}
        >
          Edit
        </button>
         
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
    </tr>
    <tr>
      <td>
        2
      </td>
      <td>
        <input
          onChange={[Function]}
          type="text"
          value="first2"
        />
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          View
        </button>
         
        <button
          onClick={[Function]}
        >
          Edit
        </button>
         
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`EditableTable should render with 3 devs 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <td>
        Id
      </td>
      <td>
        Name
      </td>
      <td>
        Action
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        1
      </td>
      <td>
        <input
          onChange={[Function]}
          type="text"
          value="first1"
        />
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          View
        </button>
         
        <button
          onClick={[Function]}
        >
          Edit
        </button>
         
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
    </tr>
    <tr>
      <td>
        2
      </td>
      <td>
        <input
          onChange={[Function]}
          type="text"
          value="first2"
        />
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          View
        </button>
         
        <button
          onClick={[Function]}
        >
          Edit
        </button>
         
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
    </tr>
    <tr>
      <td>
        3
      </td>
      <td>
        <input
          onChange={[Function]}
          type="text"
          value="first3"
        />
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          View
        </button>
         
        <button
          onClick={[Function]}
        >
          Edit
        </button>
         
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
